django/db/models/fields/__init__.py

```diff
@@ -1522,6 +1522,9 @@ def get_internal_type(self):
     def to_python(self, value):
         if value is None:
             return value
+        context = decimal.Context(prec=self.max_digits)
```

Comment:
Although fast, it's an order of magnitude slower than attribute access:

```python
In [1]: from decimal import Context

In [2]: %timeit Context(prec=1)
The slowest run took 29.15 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 613 ns per loop

In [3]: %timeit dict.update
The slowest run took 19.66 times longer than the fastest. This could mean that an intermediate result is being cached.
10000000 loops, best of 3: 53 ns per loop
```

Not to mention the poor garbage collector!